### PR TITLE
add smaller test model, refactor tests

### PR DESCRIPTION
**Pull request by iantaylor-NOAA**
This Pull Request includes the following:
- adds a new "simple_small" model
  - based on "simple"
  - has fewer years (main time series is 2011-2022)
  - minor changes to model configuration for simplicity or to implement more common settings
  - files are about 2 MB which should be small enough to keep the package within the 5 MB limit on CRAN
  - changes from "simple" made via script attached to this comment https://github.com/r4ss/r4ss/issues/206#issuecomment-1157078047
- refactors the tests to put all tests associated with 3.24 and earlier 3.30 versions into a single `test-older_ss3.R` file which can be excluded from future CRAN submissions along with those model files
- other tests should work on CRAN with the exception of those that require the executable which will be skipped

I think this draft pull request will trigger a code coverage check which will help me see if I've missed some important pieces.
Feel free to comment on any of this.

### CODE REVIEW

**Comment by k-doering-NOAA** [comment]
@iantaylor-NOAA, good call on making a smaller model. I think we should go ahead and replace any of the simple 3.30 models with this model!

I'm curious, do you think we should retain a 3.24 model for testing (even if that testing is skipped on CRAN)? And if so, should we try to make it smaller like this one?

**Comment by k-doering-NOAA** [comment]
Also, I took a brief glance through the diffs and the changes make sense to me. I haven't scrutinized everything line-by-line, though, as I figure I can do that when this is no longer a draft. Happy to take a closer look if you would like, though!

**Comment by iantaylor-NOAA** [comment]
@k-doering-NOAA, good questions. I thought about removing the old models, but it doesn't make sense to maintain the read/write functions for 3.24 models if we don't maintain the tests that check those functions.

We had a discussion about backwards compatibility in #306 and I think this PR is in line with the views shared there: maintaining tools for accessing and understanding models associated with old versions of SS3 (`SS_output()`, `SS_plots()`, and the read/write functions for the input files), but no longer testing things like the profile, retrospective, or jitter functions for old versions.

However, it surely doesn't make sense to include the files associated with those older version as part of the package that all users get when installing r4ss from github. Would it be possible to create a new repo with the simple models from the older versions of SS3 and revise the r4ss-extra-tests workflow to copy them for testing purposes? That would also save us from removing those files from future CRAN submissions.

**Comment by k-doering-NOAA** [comment]
@iantaylor-NOAA , should the simple_3.30.xx models in inst/extdata be removed before merging in, since the smaller test can serve as a replacement?

File: tests/testthat/test-control.R [hunk 1]
```diff
@@ -13,242 +13,108 @@ file.copy(example_path, tmp_path, recursive = TRUE)
 on.exit(unlink(tmp_path, recursive = TRUE))
 
 # paths to the model files
-sim_3.30.13 <- file.path(tmp_path, "extdata", "simple_3.30.13")
-sim_3.24 <- file.path(tmp_path, "extdata", "simple_3.24")
+sim_small <- file.path(tmp_path, "extdata", "simple_small")
 
-test_that("SS_readctl and SS_writectl works for 3.30.13", {
```
Comment (k-doering-NOAA) [inline tests/testthat/test-control.R:19]: I think the other 3.30.xx version tests can be removed in this PR.

File: tests/testthat/test_Sensi_plot.R [hunk 2]
```diff
@@ -59,7 +59,7 @@ test_that("SS_Sensi_plot runs", {
   SS_Sensi_plot(
     model.summaries = model.summaries,
     dir = temp_path,
-    current.year = 2001,
+    current.year = 2022,
```
Comment (k-doering-NOAA) [inline tests/testthat/test_Sensi_plot.R:62]: Good idea to make the last model year 2022 instead of 2001!

**Review by k-doering-NOAA — CHANGES_REQUESTED** [review]
Hi @iantaylor-NOAA , it looks good. I asked a few questions for you to consider. Once done with changes, I will change the workflows to reference `@main` instead .

File: tests/testthat/test-control.R [hunk 3]
```diff
@@ -13,242 +13,108 @@ file.copy(example_path, tmp_path, recursive = TRUE)
...
-    ctllist = ctl_3.24,
-    version = "3.24",
-    verbose = FALSE,
-    overwrite = TRUE,
-    outfile = file.path(sim_3.24, "testctl.ss")
-  )
-  expect_true(file.exists(file.path(sim_3.24, "testctl.ss")))
-})
-
-test_that("SS_readctl and SS_writectl works for 3.24 when datlist = FALSE", {
```
[the remark this answers is not in the document]
Comment (iantaylor-NOAA) [reply tests/testthat/test-control.R:157]: Tests removed as discussed in #702

**Comment by iantaylor-NOAA** [comment]
After way to many typos and omissions, I think I've got these changes cleaned up and the suggestions from @k-doering-NOAA implemented and tests passing, so will merge this branch.

There remains cleanup to deprecate and then remove the 3.24-related functions in #702 but that can be separate and happen later. Also, we may still want to move the 3.24 model and earlier 3.30 models used for testing `SS_output()` into a different repo and create a new github action to use it for testing.

Code coverage has decreased because the read/write functions for 3.24 models aren't being tested, and some of the elements of `SS_output()` that are specific to earlier 3.30 versions are also being skipped. But I'm OK with all those reductions.

**Comment by k-doering-NOAA** [comment]
Thanks @iantaylor-NOAA ! I can work on deprecating the 3.24 functions.

**Pull request merged**